pkg/telemetry/client.go

```diff
@@ -139,6 +139,13 @@ func (tc *telemetryClient) SaveMetrics() error {
 		return errors.Wrap(err, "unable to create the telemetry schema")
 	}
 
+	// If user opts-out from CEIP, clear the metrics data from the local database
+	ceipOptInConfigVal, _ := configlib.GetCEIPOptIn()
+	optIn, _ := strconv.ParseBool(ceipOptInConfigVal)
+	if !optIn {
+		return tc.metricsDB.ClearMetricData()
```

Comment:
nit: once emptied, subsequent attempts to delete rows from an empty table feels rather unsatisfactory. It also means possibly returning errors (failure to read/write the db) when there shouldn't be reasons for these errors to occur in the first place. Not sure if there is a way to avoid this without setting some state somewhere to indicated that the DB is wiped. If we elect not to optimize, perhaps add a TODO comment to explain.

Reply:
Agree. One possible solution/optimization could be to let the telemetry plugin to empty the DB as soon
as the user updates the opt-out choice (through telemetry plugin) and CLI would just ignore saving the metrics if user opts-out. I am hoping the release with this change would clear all the existing local metrics DB and future enhancements in telemetry plugin stated above would then be suffice.  I added TODO comment for the same.